File: backend/team6/src/test/java/programmers/team6/support/TestVacationInfoBuilder.java
package programmers.team6.support;

import programmers.team6.domain.vacation.entity.VacationInfo;

public class TestVacationInfoBuilder {

	private double totalCount;
	private String vacationType;
	private Long memberId;

	public TestVacationInfoBuilder totalCount(double totalCount) {
		this.totalCount = totalCount;
		return this;
	}

	public TestVacationInfoBuilder vacationType(String vacationType) {
		this.vacationType = vacationType;
		return this;
	}

	public TestVacationInfoBuilder memberId(Long memberId) {
		this.memberId = memberId;
		return this;
	}

	public VacationInfo build() {
		return new VacationInfo(totalCount, vacationType, memberId);
	}
}

File: backend/team6/src/test/java/programmers/team6/support/TestVacationInfoBuilder.kt
package programmers.team6.support

import programmers.team6.domain.vacation.entity.VacationInfo

class TestVacationInfoBuilder {
    private var totalCount: Double = 0.0
    private lateinit var vacationType: String
    private var memberId: Long = 0L

    fun totalCount(totalCount: Double): TestVacationInfoBuilder {
        this.totalCount = totalCount
        return this
    }

    fun vacationType(vacationType: String): TestVacationInfoBuilder {
        this.vacationType = vacationType
        return this
    }

    fun memberId(memberId: Long): TestVacationInfoBuilder {
        this.memberId = memberId
        return this
    }

    fun build(): VacationInfo {
        return VacationInfo(totalCount, vacationType, memberId)
    }
}
